### ISSUE TITLE
[comment-module] Mentions do not work in comments outside forum

### ISSUE THREAD

**Issue by emergingdzns** [pyrocms/pyrocms#4877]
**Describe the bug**
When you include the comments module on a site attached to custom streams, when the user attempts to tag other users in the comment, the user selector doesn't appear like it does in the forum module. You can't select a particular user.

**Comment by RyanThompson** [comment]
Are you including the scripts? There's a mention.js that is used.. I am updating that addon now though so I'll be cleaning some stuff up.

**Comment by emergingdzns** [comment]
Ok so I added the references to all three js files like this:
```
{{ asset_add("theme.js", "anomaly.module.comments::js/jquery.atwho.min.js") }}
{{ asset_add("theme.js", "anomaly.module.comments::js/jquery.caret.min.js") }}
{{ asset_add("theme.js", "anomaly.module.comments::js/mention.js") }}
```

I had to add them all to avoid the errors for missing functions.

I don't get any errors but I also don't get any drop down showing up when I begin typing an @ symbol and the first few letters of a name. I checked inspector and I don't see any ajax calls being made to look for matches either.

**Comment by emergingdzns** [comment]
I think I just answered my own question to a degree. I just noticed that the mention.js refers specifically to a textarea. That sucks. Our entire design was going for a more facebook like look with a single line text field in a bootstrap input group.

**Comment by emergingdzns** [comment]
I changed it to a textarea and added `data-field="content"` to the tag. I can see that mentions.js is in the source of the theme.js now but when I try to start tagging someone it still doesn't trigger the atwho function at all. I tried adding console logging to the mention js but it never outputs anything. I'm at a loss to explain it.

**Comment by emergingdzns** [comment]
Ok nevermind. I figured out the problem. The comments were being loaded via ajax. The js in mentions.js is not setup to work on dynamically added elements. I copied the content of mentions.js and plugged it into my theme initialization script like this:
```
$(document).on('focus','.comment-text',function() {
    $(this).atwho({
        at: "@",
        callbacks: {
            remoteFilter: function (query, callback) {

                if (query.length < 3) {
                    return;
                }

                $.getJSON(
                    REQUEST_ROOT_PATH + "/comments/mention",
                    {q: query},
                    function (data) {
                        callback(data)
                    }
                );
            }
        }
    });
});
```
This also allowed me to change it from being a textarea to being loaded on any form element with the class of 'comment-text'. Works great now.
